### ISSUE TITLE
Implement batch processing

### ISSUE THREAD

**Issue by benjaminLuoHC** [hc-sc-ocdo-bdpd/file-processing-tools#136]
Related to applications involving large directories mentioned in Issue #134, is the need to decrease runtime and improve fault tolerance. This can be achieved by batch processing and writing to several CSV files (then concatenating them afterward) instead of having a single program run for hours and output to a single CSV.

If the program should fail, it must also output an identifier (ex index) to pinpoint the source of error and allow execution onward from that point. This would also entail all the data prior to the failure be logged to a CSV file prior to the program throwing the error.

**Comment by benjaminLuoHC** [comment]
It might be possible to use the `threading` or `multiprocessing` library to run the program concurrently

**Comment by benjaminLuoHC** [comment]
I experimented with the 3 aforementioned libraries but saw no performance gain. The batching took ~0.02 seconds to run for the `./tests/resources` directory

```py
def dataset(file_path):
    for dirpath, _, filenames in os.walk(file_path):
        for filename in filenames:
            yield os.path.join(dirpath, filename)

def dataloader(data_generator, batch_size):
    data = list(data_generator)
    return [data[i:i + batch_size] for i in range(0, len(data), batch_size)]

batches = dataloader(dataset('./tests/resources'), batch_size=18) 
```

Threading/multi-processing with 8 CPU cores took the same amount of time (or longer) as running the program normally. This suggests that Python might already do some kind of multi-threading in the background

However, I monitored execution via the Performance/CPU tab in Task Manager. The threading/multi-processor experiments indeed utilized all 8 cores initially (1st image) before reducing decreasing as processes completed. On the other hand, the regular program only used 1 core and had a roughly constant CPU utilization (2nd image) so in theory, the experimental code should have had a greatly reduced runtime

![image](https://github.com/hc-sc-ocdo-bdpd/file-processing-tools/assets/155466242/d5687449-ebbc-4f77-a6d2-a393803b548e)

![image](https://github.com/hc-sc-ocdo-bdpd/file-processing-tools/assets/155466242/436c31ca-88ad-4c3c-9b66-28c70ace1790)

Another possible cause is the setup time for creating threads/processes causes runtime to be longer than the control case for smaller directories. If this is true, then runtime would decrease for larger directories
